subprojects/core/src/main/java/org/gradle/execution/ProjectExecutionServiceRegistry.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.internal.service.ServiceRegistry;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Registry of services provided for already configured projects.
+ */
+public class ProjectExecutionServiceRegistry {
+    private final LoadingCache<ProjectInternal, ServiceRegistry> projectRegistries = CacheBuilder.newBuilder()
```

Comment:
When do the entries get evicted?  Are they ever?

How/when is that service registry closed?

Reply:
`CacheBuilder` by default creates something that is not really a cache but a memoizer, so there's no eviction. That's what we want here.

I'm not sure what you mean by closing the service registry?

Reply:
`DefaultServiceRegistry` is a closeable which stops all services when closed. So the cache contains many closeables which would need to be closed when the build finishes. Does this happen somewhat automatic or do we need to do something to make this happen?

Reply:
Damn, I don't know how many times have I ended up opening `DefaultSerializerRegistry` instead of `DefaultServiceRegistry` in IntelliJ. You're right, fixed.